erpnext/stock/report/stock_balance/stock_balance.py

```diff
@@ -198,7 +198,9 @@ def get_item_warehouse_map(filters, sle):
 
 		value_diff = flt(d.stock_value_difference)
 
-		if d.posting_date < from_date:
+		if d.posting_date < from_date or (d.posting_date == from_date
+			and d.voucher_type == 'Stock Reconciliation' and
+			frappe.db.get_value('Stock Reconciliation', d.voucher_no, ['purpose']) == 'Opening Stock'):
```

Comment:
This seems like correct behaviour now 👍 

As discussed, better to do this in the Stock Ledger report too. 

Closing other PRs, as it would be easier to port once this is final and merged.